src/index.ts

```diff
@@ -1,21 +1,24 @@
+/* eslint-disable no-param-reassign */
```

Comment:
All the code in this file seems long and complex enough to warrant a separate file and a set of tests.